packages/backend/src/queryBuilder.ts

```diff
@@ -393,7 +509,11 @@ export const buildQuery = ({
             sqlGroupBy,
         ].join('\n');
         const cteName = 'metrics';
-        const cte = `WITH ${cteName} AS (\n${cteSql}\n)`;
+        const ctes = [
```

Comment:
Potential things to refactor:

This is a special edgecase that forces us to wrap the whole query into a CTE so we can use table calculations.

This makes me think we should probably have a better way to join CTEs in the future.